tardis/widgets/base.py

```diff
@@ -0,0 +1,546 @@
+            Default value `None` indicates that there is no changable column.
+
+        Returns
+        -------
+        qgrid.QgridWidget
+            Table widget object
+        """
+        # Setting the options to be used for creating table widgets
+        grid_options = {
+            'sortable': False,
+            'filterable': False,
+            'editable': False,
+            'minVisibleRows': 2
+        }
+        column_options = {
+            'minWidth': None,
+        }
+
+        # Check whether passed col_widths list is correct or not
+        if len(col_widths) != data.shape[1]+1:
+            raise ValueError('Size of column widths list do not match with '
+                             'number of columns + 1 (index) in dataframe')
+
+        # Note: Since forceFitColumns is enabled by default in grid_options,
+        # the column widths (when all specified) get applied in proportions,
+        # despite their original unit is px thus it's better they sum to 100
+        if sum(col_widths) != 100:
+            raise ValueError('Column widths are not proportions of 100 (i.e. '
+                             'they do not sum to 100)')
+
+        # Preparing dictionary that defines column widths
+        cols_with_index = [data.index.name] + data.columns.to_list()
+        column_widths_definitions = {col_name: {'width': col_width}
+                                     for col_name, col_width in zip(
+                                         cols_with_index, col_widths)}
+
+        # We also need to define widths for different names of changeable column
+        if changeable_col:
+            if {'index', 'other_names'}.issubset(set(changeable_col.keys())):
+                column_widths_definitions.update(
+                    {col_name: {'width': col_widths[changeable_col['index']]}
+                     for col_name in changeable_col['other_names']})
+            else:
+                raise ValueError("Changeable column dictionary does not contain "
+                                 "'index' or 'other_names' key")
+
+        # Create the table widget using qgrid
+        return qgrid.show_grid(data,
+                               grid_options=grid_options,
+                               column_options=column_options,
+                               column_definitions=column_widths_definitions)
+
+    def update_element_count_table(self, event, qgrid_widget):
+        """Event listener to update the data in element count table widget based
+        on interaction (row selected event) in shells table widget.
+
+        Parameters
+        ----------
+        event : dict
+            Dictionary that holds information about event (see Notes section)
+        qgrid_widget : qgrid.QgridWidget
+            QgridWidget instance that fired the event (see Notes section)
+
+        Note
+        ----
+        You will never need to pass any of these arguments explicitly. This is
```

Comment:
nice!

Reply:
Thanks - but for some reason, when building docs, it isn't appearing even though it is valid as per numpydoc - I'll check it out, might has to do something with sphinx configuration!

Reply:
It works now as I fixed some mistakes in docstrings.